test/integration/security-rules.spec.ts

```diff
@@ -257,26 +276,27 @@ describe('admin.securityRules', () => {
 
   describe('deleteRuleset()', () => {
     it('rejects with not-found when the Ruleset does not exist', () => {
-      const name = 'e1212' + testRuleset.name.substring(5);
-      return admin.securityRules().deleteRuleset(name)
+      const uuid = '00000000-1111-2222-3333-444444444444';
+      return admin.securityRules().deleteRuleset(uuid)
         .should.eventually.be.rejected.and.have.property('code', 'security-rules/not-found');
     });
 
     it('rejects with invalid-argument when the Ruleset name is invalid', () => {
-      return admin.securityRules().deleteRuleset('invalid')
+      return admin.securityRules().deleteRuleset('invalid uuid')
         .should.eventually.be.rejected.and.have.property('code', 'security-rules/invalid-argument');
     });
 
     it('deletes existing Ruleset', () => {
-      return admin.securityRules().deleteRuleset(testRuleset.name)
-        .then(() => {
-          return admin.securityRules().getRuleset(testRuleset.name)
-            .should.eventually.be.rejected.and.have.property('code', 'security-rules/not-found');
-        })
-        .then(() => {
-          unscheduleForDelete(testRuleset); // Already deleted.
-          testRuleset = null;
-        });
+      return createTemporaryRuleset().then((ruleset) => {
+        return admin.securityRules().deleteRuleset(ruleset.name)
+          .then(() => {
+            return admin.securityRules().getRuleset(ruleset.name)
+              .should.eventually.be.rejected.and.have.property('code', 'security-rules/not-found');
+          })
+          .then(() => {
+            unscheduleForDelete(ruleset); // Already deleted.
```

Comment:
Is the `testRuleset` variable still referenced anywhere? If not lets remove that.

Reply:
> Is the testRuleset variable still referenced anywhere?

No.

> If not lets remove that.

Remove the testRuleset? It's already been removed...? Or were you referring to the unscheduleForDelete call? That call is currently necessary since the deleteTempRulesets() will otherwise error out due to not being able to find the entity. We could change that to make it idempotent though. (Let me know if you'd like me to do that.)

Reply:
Ah, I didn't spot that the variable has already been removed. Thanks.